packages/extension-ui/src/components/InputWithLabel.tsx

```diff
@@ -39,6 +42,19 @@ function InputWithLabel ({ className, defaultValue, disabled, isError, isFocused
     [onChange]
   );
 
+  const onKeyDown = useCallback(
+    (keyEvent: React.KeyboardEvent<HTMLInputElement>) => {
+      if (type === 'password') {
+        if (keyEvent.getModifierState('CapsLock')) {
+          setIsCapsLock(true);
+        } else {
+          setIsCapsLock(false);
+        }
+      }
+    },
+    [type]
+  );
```

Comment:
This is a bit weird now since we have both on press and on down events. (Enter & capslock checking)

Reply:
Totally, I missed the elephant in the room after spending a while checking that I hadn't broken any of the wanings. That's fixed now.